### PR TITLE
Launchpad - Use launchpad endpoint to retrieve latest data

### PR DESCRIPTION
**Pull request by agrullon95**
Related to https://github.com/Automattic/wp-calypso/issues/73830

## Testing / Review

Testing: Medium
Review: Medium

## Proposed Changes

* Use @mattwiebe's launchpad endpoint hook (https://github.com/Automattic/wp-calypso/pull/73833) to retrieve the latest launchpad data (site_intent, checklist task statuses, and launchpad_screen) on the Launchpad screen.

## Testing Instructions

* Checkout this branch
* Create a launchpad-enabled site or use an existing site  (Test on an atomic site as well)
* Verify that you are not redirected to `/home` if launchpad is enabled (site option `launchpad_screen` = `full)
* Verify the launchpad checklist statuses are updating properly after completing a task
* Verify the test cases are passing by executing: `yarn run test-client client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/test`
* Verify you are redirected to home if the `launchpad_screen` site option is `off` or false

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by edanzer** [comment]
This change obviously touches a lot of files and most launchpad functionality. But for the scope, the changes look straightforward. Nice to see us using React Query here. 

I tested by running through multiple flows and confirmed everything is working correctly for each, including checking that Launchpad is loading when it should and not loading when it shouldn't, and that various task statuses update as expected as I work through each checklist. 

- [x] Test Newsletter flow
- [x] Test LIB flow
- [x] Test Free flow
- [x] Test Write flow
- [x] Test Build flow
- [x] All unit tests still pass

There are some merge conflicts to resolve, so I'll plan to do another test and sign off after those are resolved.

File: client/data/sites/use-launchpad.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export const useLaunchpad = ( siteSlug: string | null, cache = true ) => {
+	const key = [ 'launchpad', siteSlug ];
+	return useQuery( key, () => siteSlug && fetchLaunchpad( siteSlug ), {
+		meta: {
+			persist: cache,
+		},
+		enabled: true,
+		placeholderData: {
+			checklist_statuses: {},
```
Comment (mattwiebe) [inline client/data/sites/use-launchpad.ts]: Let's make this an array `[]`

**Review by edanzer — APPROVED** [review]
Latest code changes looking good. And just retested all flows, including one on Atomic. All Launchpad screen and task status are working and updating as expected. 

- [x] Tested Free flow
- [x] Tested Newsletter flow
- [x] Tested Link in Bio flow
- [x] Tested Build flow
- [x] Tested VideoPress flow
- [x] Tested Write flow on Atomic
- [x] Confirmed all tests are passing

Only issue I noticed is that on Atomic, on the first load of Launchpad, I see the Coming Soon screen. That's an existing issue, not related to to PR. I note it because one of the main points of introducing the new end point is hopefully to resolve lagging state issues on Atomic. That'll be another one we still need to address.

**Review by sharpirate — APPROVED** [review]
### Testing
✔️ Tested all flows on Simple / Atomic
✔️ All tests are passing

**Review by mattwiebe — APPROVED** [review]
Thanks for taking this over the finish line! The code looks good and tests properly for me. 

And great work on the testing integration, TIL about how to use `nock`. 👍

**Pull request merged**